src/actions/BulkSetUnit.php

```
@@ -0,0 +1,43 @@
+        /** @var ServerQuery $query */
+        $query = $this->getDataProvider()->query;
+        $query->withHardwareSettings();
+    }
+
+    public function beforeSave(): void
+    {
+        /** @var \hipanel\actions\Action $action */
+        $action = $this->controller->action;
+        $switches = Yii::$app->request->post('HardwareSettings');
+        $units = ArrayHelper::remove($switches, 'units');
+        foreach ($switches as $id => $switch) {
```

PHPMD (2.15.0)

Avoid unused local variables such as '$switch'. (Unused Code Rules)

(UnusedLocalVariable)